core/src/main/java/org/eclipse/hono/auth/Device.java

```diff
@@ -22,23 +22,18 @@
 import io.jsonwebtoken.Claims;
 import io.jsonwebtoken.Jws;
 import io.jsonwebtoken.RequiredTypeException;
-import io.vertx.core.AsyncResult;
-import io.vertx.core.Future;
-import io.vertx.core.Handler;
 import io.vertx.core.json.JsonObject;
-import io.vertx.ext.auth.AuthProvider;
-import io.vertx.ext.auth.User;
 
 
 /**
  * An authenticated client of a protocol adapter representing a device.
  * <p>
  * The device's identity and authorities are contained in a Java Web Token.
  */
-public class Device implements User {
+public class Device {
 
     private final JsonObject principal;
-    private final Set<Object> authorities = new HashSet<>();
+    protected final Set<Object> authorities = new HashSet<>();
```

Comment:
I would rather not give subclasses access to this field because it contains the security relevant information that should be immutable once created. Instead, I propose to add a method for checking if the device has a particular authority here in this class. Subclasses can then use this method for implementing e.g. the `User` interface.

Reply:
:+1: will change this.